### PR TITLE
Add missing collection methods

### PR DESCRIPTION
**Pull request by mandrean**
This PR adds missing collection methods, most importantly `Iterable<T>.toVavrSet()`, but also some others that might for example have a `.toVavrList()` but not one for `Set` etc.

It also improves the kdocs and does some general cleanup using `ktlint`.

Best reviewed commit by commit, as they are very atomic and descriptive in what they do.

### CODE REVIEW

File: src/main/kotlin/io/vavr/kotlin/Collections.kt [hunk 1]
```diff
@@ -31,121 +38,197 @@ import io.vavr.collection.Stream
...
+ * @see HashSet.ofAll
  */
-fun <T> hashSet(vararg t: T):
-        io.vavr.collection.HashSet<T> = io.vavr.collection.HashSet.ofAll(t.asIterable())
+fun <T> Array<T>.toVavrSet():
+    HashSet<T> = HashSet.ofAll(this.asIterable())
 
 /**
- * Creates a Vavr LinkedHashSet
+ * Converts a Kotlin [Sequence] into a Vavr [List].
```
Comment (ruslansennov) [inline src/main/kotlin/io/vavr/kotlin/Collections.kt]: into Set

File: src/main/kotlin/io/vavr/kotlin/Collections.kt [hunk 2]
```diff
@@ -31,121 +38,197 @@ import io.vavr.collection.Stream
...
- * Notice that the elements of a TreeSet must be Comparable
+ * Converts a Kotlin [Sequence] into a Vavr [LinkedHashSet].
+ *
+ * @see LinkedHashSet.ofAll
+ */
+fun <T> Sequence<T>.toVavrLinkedHashSet():
+    LinkedHashSet<T> = LinkedHashSet.ofAll(this.asIterable())
+
+/**
+ * Creates a Vavr [TreeSet] from a series of Kotlin [Pair]s.
```
Comment (ruslansennov) [inline src/main/kotlin/io/vavr/kotlin/Collections.kt]: of the given elements

**Review by ruslansennov — CHANGES_REQUESTED** [review]
Thanks @mandrean 
Just two comments on the kdocs

**Comment by mandrean** [comment]
Thanks for the quick feedback @ruslansennov!

Pushed a new commit fixing the typos

**Review by ruslansennov — APPROVED** [review]

**Pull request merged**